### PR TITLE
Fix HtmlReporter which was broken via a previous commit and misc

### PR DESCRIPTION
**Pull request by mach6**
- Velocity wants several classes to be public
- Also includes changes to clean up some codacy 'F' (by Files) reported code violations

### CODE REVIEW

File: client/src/main/resources/templates/css/htmlreporter.css [hunk 1]
```diff
@@ -263,7 +265,8 @@ h2.yuk_grey_midpnl_ltitle {padding:7px 9px 3px 9px; margin:0; color:#5d5d5d; fon
     width: 99%;
 }
 
-h2.yuk_grey_midpnl_ltitle {width =95%;
+h2.yuk_grey_midpnl_ltitle {
+    width: 95%;
     padding: 7px 9px 3px 9px;
```
Comment (mach6) [inline client/src/main/resources/templates/css/htmlreporter.css]: Codacy Issue found: [Using width with padding can sometimes make elements larger than you expect.](https://www.codacy.com/app/SeLion/paypal-SeLion/file/1249607196/issues/source?bid=2928425&fileBranchId=2937920#l270)

File: client/src/main/resources/templates/css/htmlreporter.css [hunk 2]
```diff
@@ -112,7 +108,18 @@ h2.yuk_grad_ltitle_skipped {
     text-align: left;
 }
 
-h2.yuk_grey_midpnl_ltitle {padding:7px 9px 3px 9px; margin:0; color:#5d5d5d; font-size:small; font-weight:bold; background-color:#e8e8e8; border-style:solid; border-color:#ccc; border-width:1px; font-family:Arial, Helvetica, sans-serif;}
+h2.yuk_grey_midpnl_ltitle {
```
Comment (mach6) [inline client/src/main/resources/templates/css/htmlreporter.css]: Codacy Issue found: [Heading (h2) has already been defined.](https://www.codacy.com/app/SeLion/paypal-SeLion/file/1249607196/issues/source?bid=2928425&fileBranchId=2937920#l111)

File: client/src/main/java/com/paypal/selion/internal/reports/runtimereport/MethodInfo.java [hunk 3]
```diff
@@ -28,23 +27,24 @@
...
 class MethodInfo {
 
     private static SimpleLogger logger = SeLionLogger.getLogger();
 
-    private String suite;
-    private String test;
-    private String packageInfo;
-    private String className;
-    private String methodName;
+    private final String suite;
```
Comment (mach6) [inline client/src/main/java/com/paypal/selion/internal/reports/runtimereport/MethodInfo.java]: Codacy Issue found: [Perhaps 'suite' could be replaced by a local variable.](https://www.codacy.com/app/SeLion/paypal-SeLion/file/1249607326/issues/source?bid=2928425&fileBranchId=2937920#l43)

File: client/src/main/java/com/paypal/selion/internal/reports/runtimereport/MethodInfo.java [hunk 4]
```diff
@@ -28,23 +27,24 @@
...
+    private final String test;
```
Comment (mach6) [inline client/src/main/java/com/paypal/selion/internal/reports/runtimereport/MethodInfo.java]: Codacy Issue found: [Perhaps 'test' could be replaced by a local variable.](https://www.codacy.com/app/SeLion/paypal-SeLion/file/1249607326/issues/source?bid=2928425&fileBranchId=2937920#l44)

File: client/src/main/java/com/paypal/selion/internal/reports/runtimereport/MethodInfo.java [hunk 5]
```diff
@@ -28,23 +27,24 @@
...
+    private final String packageInfo;
+    private final String className;
+    private final String methodName;
```
Comment (mach6) [inline client/src/main/java/com/paypal/selion/internal/reports/runtimereport/MethodInfo.java]: Codacy Issue found: [Perhaps 'methodName' could be replaced by a local variable.](https://www.codacy.com/app/SeLion/paypal-SeLion/file/1249607326/issues/source?bid=2928425&fileBranchId=2937920#l47)

File: client/src/main/java/com/paypal/selion/internal/reports/runtimereport/MethodInfo.java [hunk 6]
```diff
@@ -28,23 +27,24 @@
...
 
-    private String suite;
-    private String test;
-    private String packageInfo;
-    private String className;
-    private String methodName;
+    private final String suite;
+    private final String test;
+    private final String packageInfo;
+    private final String className;
```
Comment (mach6) [inline client/src/main/java/com/paypal/selion/internal/reports/runtimereport/MethodInfo.java]: Codacy Issue found: [Perhaps 'className' could be replaced by a local variable.](https://www.codacy.com/app/SeLion/paypal-SeLion/file/1249607326/issues/source?bid=2928425&fileBranchId=2937920#l46)

File: client/src/main/java/com/paypal/selion/internal/reports/runtimereport/MethodInfo.java [hunk 7]
```diff
@@ -28,23 +27,24 @@
...
     private static SimpleLogger logger = SeLionLogger.getLogger();
 
-    private String suite;
-    private String test;
-    private String packageInfo;
-    private String className;
-    private String methodName;
+    private final String suite;
+    private final String test;
+    private final String packageInfo;
```
Comment (mach6) [inline client/src/main/java/com/paypal/selion/internal/reports/runtimereport/MethodInfo.java]: Codacy Issue found: [Perhaps 'packageInfo' could be replaced by a local variable.](https://www.codacy.com/app/SeLion/paypal-SeLion/file/1249607326/issues/source?bid=2928425&fileBranchId=2937920#l45)

File: dataproviders/src/test/java/com/paypal/selion/platform/dataprovider/YamlDataProviderTest.java [hunk 8]
```diff
@@ -424,16 +420,6 @@ public void testGetDataByIndex_NullData() throws IOException {
...
-    public synchronized List<String> transferDataIntoList(Hashtable<String, Object> allUsers) {
-        List<String> fetchedNames = new ArrayList<String>();
-        Set<String> keys = allUsers.keySet();
-        for (String key : keys) {
-            USER user = (USER) allUsers.get(key);
-            fetchedNames.add(user.getName());
-        }
-        return fetchedNames;
-    }
-
```
Comment (mach6) [inline dataproviders/src/test/java/com/paypal/selion/platform/dataprovider/YamlDataProviderTest.java]: Codacy Issue found: [Consider using varargs for methods or constructors which take an array the last parameter.](https://www.codacy.com/app/SeLion/paypal-SeLion/file/1249607615/issues/source?bid=2928425&fileBranchId=2937920#l423)

File: client/src/main/resources/templates/css/htmlreporter.css (hunk 1 above)
Comment (mach6) [inline client/src/main/resources/templates/css/htmlreporter.css:270]: will not fix

File: client/src/main/resources/templates/css/htmlreporter.css (hunk 2 above)
Comment (mach6) [inline client/src/main/resources/templates/css/htmlreporter.css:111]: will not fix

File: client/src/main/java/com/paypal/selion/internal/reports/runtimereport/MethodInfo.java (hunk 3 above)
Comment (mach6) [inline client/src/main/java/com/paypal/selion/internal/reports/runtimereport/MethodInfo.java:43]: will not fix

File: client/src/main/java/com/paypal/selion/internal/reports/runtimereport/MethodInfo.java (hunk 4 above)
Comment (mach6) [inline client/src/main/java/com/paypal/selion/internal/reports/runtimereport/MethodInfo.java:44]: will not fix

File: client/src/main/java/com/paypal/selion/internal/reports/runtimereport/MethodInfo.java (hunk 5 above)
Comment (mach6) [inline client/src/main/java/com/paypal/selion/internal/reports/runtimereport/MethodInfo.java:47]: will not fix

File: client/src/main/java/com/paypal/selion/internal/reports/runtimereport/MethodInfo.java (hunk 6 above)
Comment (mach6) [inline client/src/main/java/com/paypal/selion/internal/reports/runtimereport/MethodInfo.java:46]: will not fix

File: client/src/main/java/com/paypal/selion/internal/reports/runtimereport/MethodInfo.java (hunk 7 above)
Comment (mach6) [inline client/src/main/java/com/paypal/selion/internal/reports/runtimereport/MethodInfo.java:45]: will not fix

File: dataproviders/src/test/java/com/paypal/selion/platform/dataprovider/YamlDataProviderTest.java (hunk 8 above)
Comment (mach6) [inline dataproviders/src/test/java/com/paypal/selion/platform/dataprovider/YamlDataProviderTest.java:436]: will not fix

File: client/src/main/java/com/paypal/selion/internal/reports/html/HtmlReporterListener.java [hunk 9]
```diff
@@ -357,7 +357,7 @@ private void generateTheStackTrace(Throwable exception, ITestNGMethod method, St
             StringBuilder contentBuffer) {
         logger.entering(new Object[] { exception, method, title, contentBuffer });
         contentBuffer.append(" <div class='stContainer' >" + exception.getClass() + ":" + title// escape(title)
-                + "<a  class='exceptionlnk'>(+)</a>");
+                + "<a class='exceptionlnk' herf='#'>(+)</a>");
```
Comment (sebady) [inline client/src/main/java/com/paypal/selion/internal/reports/html/HtmlReporterListener.java]: typo

File: client/src/main/java/com/paypal/selion/internal/reports/runtimereport/MethodInfo.java [hunk 10]
```diff
@@ -52,7 +52,7 @@
     private String exception;
     private String stacktrace;
     private List<LogInfo> logs;
-    private transient ITestResult result;
+    private final transient ITestResult result;
```
Comment (sebady) [inline client/src/main/java/com/paypal/selion/internal/reports/runtimereport/MethodInfo.java:57]: transient and final?  I hope codacy didn't recommend  this?

**Comment by sebady** [comment]
The changes here look pretty good.  Can we add a way to verify the html runtime report though?

File: client/src/main/java/com/paypal/selion/internal/reports/html/HtmlReporterListener.java (hunk 9 above)
Comment (mach6) [inline client/src/main/java/com/paypal/selion/internal/reports/html/HtmlReporterListener.java]: Will fix

File: client/src/main/java/com/paypal/selion/internal/reports/runtimereport/MethodInfo.java (hunk 10 above)
Comment (mach6) [inline client/src/main/java/com/paypal/selion/internal/reports/runtimereport/MethodInfo.java:57]: it did.. what's the concern? `result` gets assigned an object which should never change and is not serializable.

**Comment by mach6** [comment]
> Can we add a way to verify the html runtime report though?

I don't follow. Do you mean a functional test for the html content created by `HtmlReporterListener`?

File: client/src/main/java/com/paypal/selion/internal/reports/runtimereport/MethodInfo.java (hunk 10 above)
Comment (mach6) [inline client/src/main/java/com/paypal/selion/internal/reports/runtimereport/MethodInfo.java:57]: alternatively we can/probably should use;

``` java
@Expose(serialize = false)
private final ITestResult result
```

Comment (sebady) [inline client/src/main/java/com/paypal/selion/internal/reports/runtimereport/MethodInfo.java:57]: I think the rule in codacy about making all fields "final" can be a bit of a blunt instrument.. As I look a this though it certainly isn't "wrong" it just looks really unusual to me.  The gson \@Expose format certainly helps to clarify what's happening for developers.

**Comment by sebady** [comment]
Yes. When we perform future refactoring can we detect any problems we introduce in the Html reporter via a test failure?  Perhaps we can persist the htmlreporter html file  and have a test verify the (generated) html against it?  Anything really to exercise the functionality of HtmlReporterListener is "correct".

File: client/src/main/java/com/paypal/selion/internal/reports/runtimereport/MethodInfo.java (hunk 10 above)
Comment (mach6) [inline client/src/main/java/com/paypal/selion/internal/reports/runtimereport/MethodInfo.java:57]: sure. It does look odd at first glance.. I'll switch this to the gson annotation

**Comment by mach6** [comment]
> Yes. When we perform future refactoring can we detect any problems we introduce in the Html reporter via a test failure? Perhaps we can persist the htmlreporter html file and have a test verify the (generated) html against it? Anything really to exercise the functionality of HtmlReporterListener is "correct".

sounds like an issue for the backlog. I'm in agreement.. more test cases needed here.

**Pull request merged**